infrastructure/src/main/java/org/corfudb/infrastructure/SequencerServer.java

```diff
@@ -148,11 +147,19 @@
     private final CorfuMsgHandler handler =
             CorfuMsgHandler.generateHandler(MethodHandles.lookup(), this);
 
-
     @Getter
     @Setter
     private volatile long sequencerEpoch = Layout.INVALID_EPOCH;
 
+    /**
+     * The lower bound of the consecutive epoch range that this sequencer
+     * observes as the primary sequencer. i.e. this sequencer has been the
+     * primary sequencer for all the consecutive epochs from this epoch to
+     * {@link this#sequencerEpoch}
+     */
+    @Getter
+    private volatile long epochRangeLowerBound = Layout.INVALID_EPOCH;
```

Comment:
Please use AtomicLong instead of volatile

Reply:
And also why you need volatile if the sequencer is single threaded

Reply:
Will do.